docs/fileformat-prose.rst

```diff
@@ -0,0 +1,180 @@
+Disk representation
+-------------------
+
+.. note::
+   these docs are written for anndata 0.7.
+   Files written before this version may differ in some conventions, but will still be read by newer versions of the library.
+
+AnnData objects are saved on disk to hierarchichal array stores like `HDF5` and `zarr`.
+This allows us to have very similar structures in disk and on memory.
+
+AnnData objects can hold three kinds of objects in it’s dimensioned mappings (i.e. `X`, `obsm`, `layers` etc.).
+These are (1) dense arrays, (2) sparse arrays, and (3) data frames.
+As an example we’ll look into a typical `.h5ad` object that’s been through an analysis.
+
+.. note::
+   I’ve started using h5py since I couldn’t figure out a nice way to print attributes from bash.
+
+.. code:: python
+
+   >>> import h5py
+   >>> f = h5py.File("02_processed.h5ad", "r")
+   >>> list(f.keys())
+   ['X', 'layers', 'obs', 'obsm', 'uns', 'var', 'varm'] 
+
+.. code:: bash
+
+   $ h5ls 02_processed.h5ad
+   X                        Group
+   layers                   Group
+   obs                      Group
+   obsm                     Group
+   uns                      Group
+   var                      Group
+   varm                     Group
+
+Dense arrays
+~~~~~~~~~~~~
+
+Dense arrays have the most simple representation on disk, as they have clear equivalents in hdf5 and zarr.
+Any dense array will be written to the file as a `dataset`.
+We can see an example of this for principle components stored in the `obsm` group:
+
+.. code:: python
+
+   >>> f["obsm"].visititems(lambda k, v: print(f"{k}: {v}")) 
+   X_pca: <HDF5 dataset "X_pca": shape (38410, 50), type "<f4">
+   X_umap: <HDF5 dataset "X_umap": shape (38410, 2), type "<f4">
+
+.. code:: bash
+
+   $ h5ls 02_processed.h5ad/obsm
+   X_pca                    Dataset {38410, 50}
+   X_umap                   Dataset {38410, 2}
+
+Sparse arrays
+~~~~~~~~~~~~~
+
+Sparse arrays don’t have a native representations in hdf5 or zarr, so we use a representation as close to the in memory datasets as we can.
```

Comment:
I think that’s underselling. I think what we do is the most straightforward and obvious way to do it.